.github/workflows/build.yaml

```diff
@@ -0,0 +1,30 @@
+name: Build
+
+on:
+  pull_request:
+    branches: [ "*" ]
+  push:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 9.0.x
+
+      - name: Setup NUGET_LOCAL_PATH
```

Comment:
Is this step necessary?  If we're not compiling any of the samples, I don't think we need the `packages` directory or any of the .nupkg files.

Reply:
The http-hello sample is compiled because it is included in the solution file. I can remove that if you'd prefer we stop compiling it on `dotnet build`.

https://github.com/dicej/spin-dotnet-sdk/blob/f039873237cea146c752593b9f860ca715750dfb/Spin.SDK.sln#L10-L11

Reply:
Ah, I didn't realize the .sln file was relevant here.  I guess I thought it was just an IDE thing for Visual Studio.  

Yeah, let's add all the samples to the build unless there's a good reason not to.  Then we can add a step to smoke test them later, as well.

Reply:
I removed them from the build. Ideally we only want to compile the relevant bits necessary to compile the SDK. The samples can be compiled separately as part of a smoke test. (I'll add that as a follow-up PR).